app/src/main/java/com/everest/moviedb/model/MovieRepository.kt

```diff
@@ -0,0 +1,82 @@
+package com.everest.moviedb.model
+
+import com.everest.moviedb.callbacks.ApiResponseCallback
+import com.everest.moviedb.ui.MovieDetail
+import com.everest.moviedb.utils.IMAGE_BASE_URL
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MovieRepository constructor(
+    private val movieApiClient: MovieService,
+    private val movieDatabaseClient: MovieDatabase
+) {
+
+    fun getAllMovies(apiResponseCallback: ApiResponseCallback){
+        var movieList: List<Movie>
+        val response = movieApiClient.getAllMovies()
+
+        response.enqueue(object : Callback<APIResponse> {
+            override fun onResponse(call: Call<APIResponse>, response: Response<APIResponse>) {
+                if (response.body() != null) {
+                    apiResponseCallback.onSuccess(convertDTOIntoUIModel(response.body()!!.movies))
+                    movieList = response.body()?.movies!!
+                    movieDatabaseClient.movieDao().insertAllMovies(movieList)
+                } else {
+                    apiResponseCallback.onFailure("Error in ApiResponseCallback")
+                }
+            }
+
+            override fun onFailure(call: Call<APIResponse>, t: Throwable) {
+                movieList = movieDatabaseClient.movieDao().getPopularMovies()
+                apiResponseCallback.onSuccess(convertDTOIntoUIModel(movieList))
+            }
```

Comment:
This looks a bit awkward. in onFailure you are returning it as success. something related to naming or callback is wrong here.